### ISSUE TITLE
trap Inf/-Inf/NaN in MCEM calc_E_llk_gen$run

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#878]
In `calculate(model, latentCalcNodes)` we can get invalid logProbs (and this has happened to a couple users) when model calculation has numerical issues.

We should probably trap this case and error out and give users a way to see the current logProb value and current model values. I think if we put a nimStop inside `calc_E_llk_gen$run`, then if the user has compiled the model themself (not relying on build_MCEM to do it) then they could just query the compiled model state once they get control back.

However, I need feedback from @danielturek @NLMichaud because currently if L-BFGS-B fails, we fall back to BFGS. Presumably this is trapping a different situation -- the comment in the code is 
```
## if constraints prevent optim from running, remove constraints. 
 ## this can happen if bounds are a function of other parameters in the model,
```
so we presumably need to disentangle the Inf/-Inf/NaN case from the case we were previously considering.

Side note, in one user's case, calculate returns "-inf". Need to check if `== -Inf` will trap such cases.

**Comment by danielturek** [comment]
Sounds reasonable to me, but @NLMichaud is infinitely closer to this code and those design decisions, than myself.

**Comment by paciorek** [comment]
Addressed by PR #887
